provider/disk_utils.py

```diff
@@ -0,0 +1,139 @@
+    :param fstype: The kind of filesystem (ntfs or xfs).
+    """
+    session = vm.wait_for_login()
+    os_type = params.get("os_type")
+    image_params = params.object_params(image_name)
+    img_size = image_params.get("image_size")
+
+    if os_type == "linux":
+        # Send only the disk ID
+        disk_id = did.split("/")[-1]
+        utils_disk.create_filesyetem_linux(session, disk_id, fstype)
+        if utils_disk.mount(did, dst_dir, fstype, session=session):
+            return dst_dir
+    else:
+        utils_disk.update_windows_disk_attributes(session, did)
+        utils_disk.clean_partition_windows(session, did)
+        return utils_disk.configure_empty_disk(session, did, img_size, "windows")[0]
+
+
+def execute_io_test(
+    test,
+    params,
+    vm,
+    image,
+    serial,
+    fstype=None,
+    dst_dir=None,
+    io_command=None,
+    clean=True,
+    ignore_all_errors=False,
+    timeout=60,
+):
+    """
+    Execute the I/O write test on the disk image inside the guest.
+    Note: The target disk image should be the uninitialized disk.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param image: The image name. e.g: stg0
+    :param serial: The serial number of the disk
+    :param dst_dir: The destination directory for I/O write testing.
+    :param io_command: The command for I/O test.
+    :param clean: If True, cleans the environment
+    :param ignore_all_errors: Checks the errors of the I/O command
+    :param timeout: The timeout for the I/O command.
+    """
+    session = vm.wait_for_login()
+    is_windows = params["os_type"] == "windows"
+
+    if fstype is None:
+        fstype = "ntfs" if is_windows else "xfs"
+
+    did = get_disk_by_serial(params.get("os_type"), serial, session)
+
+    if not dst_dir:
+        dst_dir = f"{did}:\\{image}" if is_windows else f"/mnt/{image}"
+        if is_windows:
+            session.cmd(f"mkdir {dst_dir}")
+        else:
+            session.cmd(f"mkdir -p {dst_dir}")
+
+    disk = init_disk_by_id(params, vm, did, image, dst_dir, fstype)
+    try:
+        # Start to run I/O command
+        destination = f"{dst_dir}\\test.dat" if is_windows else f"{dst_dir}/test.img"
+        if io_command:
+            io_cmd = io_command % destination
+            session.cmd(io_cmd, timeout, ignore_all_errors=ignore_all_errors)
+        else:
+            test.cancel("The I/O command has not been configured")
+
+    finally:
+        if clean:
+            # We need to try to clean up and roll back the environment finally.
+            try:
+                if disk:
+                    if not is_windows:
+                        utils_disk.is_mount(
+                            disk, dst_dir, fstype=fstype, session=session
+                        )
+                        utils_disk.umount(disk, dst_dir, fstype=fstype, session=session)
+                        session.cmd(f"rm -rf {dst_dir}")
```

Comment:
follow your code , it should be disk instead of dst_dir

Reply:
Updated !

Reply:
Updated !